auth_user_basic.go

```diff
@@ -26,6 +26,7 @@ type basicUser struct {
 	Key          string   `bson:"key" json:"key" yaml:"key"`
 	AccessRoles  []string `bson:"roles" json:"roles" yaml:"roles"`
 	Invalid      bool     `bson:"invalid" json:"invalid" yaml:"invalid"`
+	roleManager  RoleManager
```

Comment:
should the constructor have a pointer to this? does user need a setter?

Reply:
oops, added